graphrag_sdk/source.py

```
@@ -38,33 +37,54 @@ def Source(path: str, instruction: str | None = None) -> "AbstractSource":
+        AbstractSource: A string source object.
+    """
+    if not isinstance(text, str) or text == "":
+        raise Exception("Invalid argument, text should be a none empty string.")
+
+    s = STRING(text)
+    s.instruction = instruction
+
+    return s
+
 
 class AbstractSource(ABC):
```

Ruff (0.8.2)

`AbstractSource` is an abstract base class, but it has no abstract methods or properties

(B024)